src/main/kotlin/interactor/GetSalaryAverageAndCitiesNamesInCountryInteractor.kt

```diff
@@ -0,0 +1,28 @@
+package interactor
+
+import model.CityEntity
+import java.io.IOException
+
+class GetSalaryAverageAndCitiesNamesInCountryInteractor(
+    private val dataSource: CostOfLivingDataSource
+    ) {
+
+    fun execute(countryName:String):List<Pair<String,Float?>> {
+        return if (checkIfCountryIsValid(countryName))
+            dataSource.getAllCitiesData()
+                .filter{excludeNullSalariesAndLowQualityData(it) && it.country.lowercase() == countryName.lowercase()}
+                .map{Pair(it.cityName,it.averageMonthlyNetSalaryAfterTax)}
+        else
+            throw IOException("Invalid country name")
```

Comment:
```suggestion
            throw Exception("Invalid country name")
```
IOExceptions just for exceptions when accessing the **files and directories**,

Reply:
done